chest/core.py

```diff
@@ -196,7 +196,12 @@ def __setitem__(self, key, value):
 
     def __del__(self):
         if self._explicitly_given_path:
-            self.flush()
+            if os.path.exists(self.path):
+                self.flush()
+            else:
```

Comment:
The only way that `else` is triggered is if the user already called `drop()`, in which case I think that its safe to delete the chest.  The alternative for that `else` would be to re-create `self.path` and then call `flush()`, which would make `drop()` then `del` a round-about way to delete the keys that are on disk while preserving the ones in memory.